.gitlab-ci.yml

```diff
@@ -25,170 +27,144 @@ stages:
   - build
   - test
 
-fetch_tests:
+
+fetch_regression_tests:
   stage: fetch
   script:
-    - echo "Fetching tests"
-    - make test-checkout-gitlab
+    - echo "Fetching regression tests"
+    - make test-checkout-regression-gitlab
+    - mkdir -p regression_tests/riscv_tests_soc
+    - cp -r regression_tests/riscv_tests/* regression_tests/riscv_tests_soc
   artifacts:
-    name: "$CI_JOB_NAME-$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA"
-    paths:
-      - tests/*
+     name: "$CI_JOB_NAME-$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA"
+     paths:
+       - regression_tests/*
 
 fetch_ips:
   stage: fetch
   script:
+    - source setup/vsim.sh
     - echo "Fetching IPs"
-    - make checkout
-    - echo "Fetching VIPs"
-    - ./rtl/vip/get-vips.sh --yes --gitlab
+    - ./update-ips
     - echo "Generate scripts with DPI and VIP support"
-    - ./generate-scripts --rt-dpi --i2c-vip --flash-vip --i2s-vip
-
+    - ./generate-scripts 
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA"
     paths:
       - ips/*
       - sim/*
       - ipstools/*
-      - rtl/vip/*
       - .cached_ipdb.json
 
-# This jobs result is too large to produce an artifact
-# fetch_sdk:
-#   stage: fetch
-#   script:
-#     - echo "Fetching SDK from releases and setting up paths"
-#     - make sdk-gitlab
-#   artifacts:
-#     paths:
-#       - pkg/
-#       - env/
 
 build_rtl:
   stage: build
   before_script:
   script:
     - echo "Compiling RTL model and DPI libraries"
     - make build
-    - echo "Installing scripts"
-    - make install
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA"
     paths:
       - tests/*
       - ips/*
       - sim/*
       - ipstools/*
-#      - install/* # we have to rebuild in test stage because rtl libraries are
-                   # not relocatable (hardcoded paths)
-      - rtl/tb/remote_bitbang/* # we want to reuse bitbang lib
-#      - pkg/* # sdk is too large
+      - rtl/*
+
 
-# Use sdk-releases to run all tests
-test_sw:
+
+# Use simplified pulp-runtime to run a subset of tests
+test_p_bare_regressions:
   stage: test
   before_script:
-    - echo "Fetching SDK from releases and setting up paths"
-    - make sdk-gitlab
-    - echo "Compiling RTL model and DPI libraries"
+    - echo "Re-build"
     - make build
-    - echo "Installing scripts"
-    - make install
+    - echo "Fetching Runtime"
+    - make pulp-runtime
+    - echo "Source and exports"
+    - source pulp-runtime/configs/pulp.sh; 
+    - export PULP_RISCV_GCC_TOOLCHAIN=/usr/pack/pulpsdk-1.0-kgf/artifactory/pulp-sdk-release/pkg/pulp_riscv_gcc/1.0.14/
+    - export PATH=/usr/pack/pulpsdk-1.0-kgf/artifactory/pulp-sdk-release/pkg/pulp_riscv_gcc/1.0.14/bin/:$PATH
```

Comment:
There is a `1.0.16` release. Any reason why you use `1.0.14`?

Reply:
No particular reason. I updated it :)